### PR TITLE
chore: add `libraries_bom_version` in generation configuration

### PR DESCRIPTION
**Pull request by JoeWang1127**
In this PR:
- Add `libraries_bom_version` in `generation_config.yaml`.
- Modify renovate config to update `libraries_bom_version`.

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -51,6 +51,16 @@
       "depNameTemplate": "com.google.cloud:libraries-bom",
       "datasourceTemplate": "maven"
     },
+    {
+      "fileMatch": [
+        "generation_config.yaml"
+      ],
+      "matchStrings": [
+        "libraries_bom_version\\s*:\\s*(?<currentValue>.*?)\\n"
```
Comment (diegomarquezp) [inline renovate.json:59]: Note: I confirmed a similar entry to update gapic_generator_java_version worked via local run of renovate.

**Review by diegomarquezp — APPROVED** [review]

File: renovate.json (hunk 1 above)
Comment (JoeWang1127) [reply renovate.json:59]: Thanks.

I also confirm this change works using
```
export LOG_LEVEL=debug && npx --yes --package renovate -- renovate --platform=local
```

Comment (blakeli0) [reply renovate.json:59]: Thanks guys! What is the process of running renovate locally? I found this [doc](https://docs.renovatebot.com/modules/platform/local/) but not sure if it covers everything.

**Review by blakeli0 — APPROVED** [review]

File: renovate.json (hunk 1 above)
Comment (JoeWang1127) [reply renovate.json:59]: The command can output what renovate will do based on the configuration.

If you checkout this branch and run the command, you will see renovate will update the version in `generation_config.yaml` using regex manager.

**Pull request merged**